apps/server/src/modules/files/uc/sync-embedded-files.uc.ts

```diff
@@ -77,21 +115,33 @@ export class SyncEmbeddedFilesUc {
 		}
 	}
 
+	private async updateTaskLinks(file: SyncFileItem) {
+		const task = await this.embeddedFilesRepo.findTask(new ObjectId(file.parentId));
+
+		if (task) {
+			task.description = this.replaceLink(task.description, file);
+			await this.embeddedFilesRepo.updateTask(task);
+		}
+	}
+
 	private async updateLessonsLinks(file: SyncFileItem) {
 		const lesson = await this.embeddedFilesRepo.findLesson(new ObjectId(file.parentId));
 
 		if (lesson) {
 			lesson.contents = lesson.contents.map((item: IComponentProperties) => {
-				const regex = new RegExp(`${fileUrlRegex}${file.source.id}.+?"`, 'g');
-				const newUrl = `"/api/v3/file/download/${file.fileRecord.id}/${file.fileRecord.name}"`;
-
 				if ('text' in item.content) {
```

Comment:
I like something like if(isLessonTextContent(item)) in a function that we have speaking condition checks. But it is nice to have.